Determine opam-repository fork user from URI

NathanReb opened the pull request:
# What's wrong

When running `dune-release opam submit`, the actual PR creation on `opam-repository` can fail due to an invalid `head` field in the API requests. As far as I understand, this appears to happen when the user or organisation owning the package repo differs from the `opam-repository` fork owner. See #60 for further details.

# The actual fix

Since `dune-release opam submit` knows the `opam-repository` fork URI, it can be used to determine the owner of the fork and pass it in the github API request instead of the package repository owner.

# Notes

This is a first draft, I mostly wanted to have your feedback on the fix.

I wasn't sure whether HTTP URIs should be supported along with SSH ones here but that should be fairly easy to add.

If the parsing of the URI fails it falls back to `user`.

You'll also note I added some tests in a separate commit.

Let me know what you think!

Thanks

NathanReb commented:
Fixed the CI in #65 !

samoht commented:
That looks great, thanks! 

A few remarks:
- the `opam_fork_user` variable is a bit confusing: I had to look at the code again to understand how it's different from the already existing`user`; so maybe the right fix would be to rename `user` to something better (maybe `distrib_user`?) to be able to use `user` instead of `opam_fork_user`. What do you think?
-  The additions of the tests are very welcome, but I would prefer if they use alcotest instead of ounit, if possible. I'll be very happy to add more tests later :-)

NathanReb commented:
I renamed the `user` and `opam_fork_user` variables and parameters and switched to `alcotest`!

I'm an `alcotest` virgin and because it lacks ounit nesting I'm torn between having a `test_case` per `check` and the current implem. I personally feel like it should be the former but without the nesting it doesn't feel so great and it makes the test executable output very verbose so I went for the latter!

samoht commented:
Looks great now, thanks!

samoht commented:
> Also realized dune-release isn't built with dune yet, happy to contribute with the upgrade as well!

That would be great!
